datafusion/optimizer/src/analyzer/inline_table_scan.rs

```diff
@@ -51,65 +47,37 @@ impl AnalyzerRule for InlineTableScan {
 }
 
 fn analyze_internal(plan: LogicalPlan) -> Result<Transformed<LogicalPlan>> {
-    match plan {
-        // Match only on scans without filter / projection / fetch
-        // Views and DataFrames won't have those added
-        // during the early stage of planning
-        LogicalPlan::TableScan(TableScan {
-            table_name,
-            source,
-            projection,
-            filters,
-            ..
-        }) if filters.is_empty() && source.get_logical_plan().is_some() => {
-            let sub_plan = source.get_logical_plan().unwrap();
-            let projection_exprs = generate_projection_expr(&projection, sub_plan)?;
-            LogicalPlanBuilder::from(sub_plan.clone())
-                .project(projection_exprs)?
-                // Ensures that the reference to the inlined table remains the
-                // same, meaning we don't have to change any of the parent nodes
-                // that reference this table.
-                .alias(table_name)?
-                .build()
-                .map(Transformed::yes)
-        }
-        LogicalPlan::Filter(filter) => {
-            let new_expr = filter.predicate.transform(&rewrite_subquery).data()?;
-            Filter::try_new(new_expr, filter.input)
-                .map(|e| Transformed::yes(LogicalPlan::Filter(e)))
+    // rewrite any subqueries in the plan first
+    let result = plan.map_subqueries(|plan| plan.transform_up(&analyze_internal))?;
```

Comment:
```suggestion
    let plan = plan.map_subqueries(|plan| plan.transform_up(&analyze_internal))?;
```

When I read "result" I'm kinda expecting this to be a `Result<_, _>` which -- based on the `result.transform_data` call below -- is not the case here.

Same comment applies to the `let result` a few lines below.

Reply:
It is a good call -- renamed to `transformed_plan` in 0e0ac8ee5